test/activation_based/test_nir_exchange.py

```
@@ -0,0 +1,293 @@
+    )
+    with pytest.raises(ValueError, match="uniform"):
+        nir_exchange.import_from_nir(graph)
+
+    heterogeneous_r = nir.NIRGraph.from_list(
+        nir.IF(
+            r=np.array([1e4, 1e4 + 0.01]),
+            v_threshold=np.ones(2),
+            v_reset=np.zeros(2),
+        )
+    )
+    with pytest.raises(ValueError, match="nir.IF.r must be uniform"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)